src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/HttpApi/ApiClient.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTests.HttpApi
+{
+    internal sealed class ApiClient : IDisposable
```

Comment:
I realize this is probably outside the scope of this PR, but do we really want to write our own api client here? Could we not generate one from the openapi spec and use that instead?

Reply:
Because of the structure of our API surface (especially the ProcessKey type and custom type converter) and the fact that OpenAPI is not good about describing streaming endpoints, the generated code from a client generation tool based on the OpenAPI document won't be very good. It'll only 100% work for /processes and /metrics. So I will probably need to hand write the rest of them anyway.